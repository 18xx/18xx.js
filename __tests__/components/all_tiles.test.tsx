import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { mapDefinition } from '../support/map_definition';

import AllTiles from '../../src/components/all_tiles';

describe('AllTiles', () => {
  describe('#render', () => {
    // This test is too slow as it renders all the tiles
    it('is pending');
  });
});
