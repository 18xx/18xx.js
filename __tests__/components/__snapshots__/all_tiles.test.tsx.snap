// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllTiles #render matches the snapshot 1`] = `
<div
  id="all-tiles"
>
  <div>
    <svg
      height={128}
      viewBox="0 0 110.85125168440814 128"
      width={110.85125168440814}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="yellow"
        points="0,32 55.42562584220407,0 110.85125168440814,32 110.85125168440814,96 55.42562584220407,128 0,96"
      />
      <path
        d="M27.712812921102028 15.999999999999986 A 96 96 0 0,0 110.85125168440814 64.00000000000001"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <path
        d="M83.13843876330606 111.99999999999997 A 96 96 0 0,0 6.49186599675415e-14 63.99999999999997"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <g>
        <circle
          cx={42.969000690499584}
          cy={35.88225099390857}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
        <circle
          cx={67.88225099390851}
          cy={92.11774900609143}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
      </g>
      <g>
        <circle
          cx={55.42562584220407}
          cy={112}
          fill="white"
          r={9}
          stroke="#000"
          strokeWidth="2"
        />
        <text
          fill="black"
          fontSize="11"
          textAnchor="middle"
          x={55.42562584220407}
          y={116}
        >
          10
        </text>
      </g>
      <text
        fill="black"
        fontSize={8}
        textAnchor="end"
        x={109.85125168440814}
        y={94}
      >
        1.0
      </text>
    </svg>
    <svg
      height={128}
      viewBox="0 0 110.85125168440814 128"
      width={110.85125168440814}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="yellow"
        points="0,32 55.42562584220407,0 110.85125168440814,32 110.85125168440814,96 55.42562584220407,128 0,96"
      />
      <path
        d="M3.681368165081119e-15 64 A 96 96 0 0,0 83.13843876330611 16.000000000000043"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <path
        d="M110.85125168440827 64 A 96 96 0 0,0 27.712812921102028 112.00000000000003"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <g>
        <circle
          cx={24.84662832984197}
          cy={60.72887932375056}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
        <circle
          cx={86.00462335456622}
          cy={67.27112067624944}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
      </g>
      <g>
        <circle
          cx={55.42562584220407}
          cy={112}
          fill="white"
          r={9}
          stroke="#000"
          strokeWidth="2"
        />
        <text
          fill="black"
          fontSize="11"
          textAnchor="middle"
          x={55.42562584220407}
          y={116}
        >
          10
        </text>
      </g>
      <text
        fill="black"
        fontSize={8}
        textAnchor="end"
        x={109.85125168440814}
        y={94}
      >
        1.1
      </text>
    </svg>
    <svg
      height={128}
      viewBox="0 0 110.85125168440814 128"
      width={110.85125168440814}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="yellow"
        points="0,32 55.42562584220407,0 110.85125168440814,32 110.85125168440814,96 55.42562584220407,128 0,96"
      />
      <path
        d="M27.712812921102014 112.00000000000003 A 96 96 0 0,0 27.712812921102042 15.999999999999986"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <path
        d="M83.1384387633061 16.00000000000005 A 96 96 0 0,0 83.13843876330607 111.99999999999994"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <g>
        <circle
          cx={37.30325348154649}
          cy={88.84662832984196}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
        <circle
          cx={73.54799820286166}
          cy={39.15337167015803}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
      </g>
      <g>
        <circle
          cx={55.42562584220407}
          cy={112}
          fill="white"
          r={9}
          stroke="#000"
          strokeWidth="2"
        />
        <text
          fill="black"
          fontSize="11"
          textAnchor="middle"
          x={55.42562584220407}
          y={116}
        >
          10
        </text>
      </g>
      <text
        fill="black"
        fontSize={8}
        textAnchor="end"
        x={109.85125168440814}
        y={94}
      >
        1.2
      </text>
    </svg>
    <svg
      height={128}
      viewBox="0 0 110.85125168440814 128"
      width={110.85125168440814}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="yellow"
        points="0,32 55.42562584220407,0 110.85125168440814,32 110.85125168440814,96 55.42562584220407,128 0,96"
      />
      <path
        d="M83.13843876330606 111.99999999999997 A 96 96 0 0,0 6.49186599675415e-14 63.99999999999997"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <path
        d="M27.712812921102028 15.999999999999986 A 96 96 0 0,0 110.85125168440814 64.00000000000001"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <g>
        <circle
          cx={67.88225099390851}
          cy={92.11774900609143}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
        <circle
          cx={42.969000690499584}
          cy={35.88225099390857}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
      </g>
      <g>
        <circle
          cx={55.42562584220407}
          cy={112}
          fill="white"
          r={9}
          stroke="#000"
          strokeWidth="2"
        />
        <text
          fill="black"
          fontSize="11"
          textAnchor="middle"
          x={55.42562584220407}
          y={116}
        >
          10
        </text>
      </g>
      <text
        fill="black"
        fontSize={8}
        textAnchor="end"
        x={109.85125168440814}
        y={94}
      >
        1.3
      </text>
    </svg>
    <svg
      height={128}
      viewBox="0 0 110.85125168440814 128"
      width={110.85125168440814}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="yellow"
        points="0,32 55.42562584220407,0 110.85125168440814,32 110.85125168440814,96 55.42562584220407,128 0,96"
      />
      <path
        d="M110.85125168440827 64 A 96 96 0 0,0 27.712812921102028 112.00000000000003"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <path
        d="M3.681368165081119e-15 64 A 96 96 0 0,0 83.13843876330611 16.000000000000043"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <g>
        <circle
          cx={86.00462335456622}
          cy={67.27112067624944}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
        <circle
          cx={24.84662832984197}
          cy={60.72887932375056}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
      </g>
      <g>
        <circle
          cx={55.42562584220407}
          cy={112}
          fill="white"
          r={9}
          stroke="#000"
          strokeWidth="2"
        />
        <text
          fill="black"
          fontSize="11"
          textAnchor="middle"
          x={55.42562584220407}
          y={116}
        >
          10
        </text>
      </g>
      <text
        fill="black"
        fontSize={8}
        textAnchor="end"
        x={109.85125168440814}
        y={94}
      >
        1.4
      </text>
    </svg>
    <svg
      height={128}
      viewBox="0 0 110.85125168440814 128"
      width={110.85125168440814}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="yellow"
        points="0,32 55.42562584220407,0 110.85125168440814,32 110.85125168440814,96 55.42562584220407,128 0,96"
      />
      <path
        d="M83.1384387633061 16.00000000000005 A 96 96 0 0,0 83.13843876330607 111.99999999999994"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <path
        d="M27.712812921102014 112.00000000000003 A 96 96 0 0,0 27.712812921102042 15.999999999999986"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <g>
        <circle
          cx={73.54799820286166}
          cy={39.15337167015803}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
        <circle
          cx={37.30325348154649}
          cy={88.84662832984196}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
      </g>
      <g>
        <circle
          cx={55.42562584220407}
          cy={112}
          fill="white"
          r={9}
          stroke="#000"
          strokeWidth="2"
        />
        <text
          fill="black"
          fontSize="11"
          textAnchor="middle"
          x={55.42562584220407}
          y={116}
        >
          10
        </text>
      </g>
      <text
        fill="black"
        fontSize={8}
        textAnchor="end"
        x={109.85125168440814}
        y={94}
      >
        1.5
      </text>
    </svg>
  </div>
  <div>
    <svg
      height={128}
      viewBox="0 0 110.85125168440814 128"
      width={110.85125168440814}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="yellow"
        points="0,32 55.42562584220407,0 110.85125168440814,32 110.85125168440814,96 55.42562584220407,128 0,96"
      />
      <line
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
        x1={110.85125168440814}
        x2={0}
        y1={64}
        y2={63.99999999999999}
      />
      <path
        d="M27.712812921102035 15.999999999999991 A 32 32 0 0,0 83.13843876330611 16.000000000000014"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <g>
        <circle
          cx={83.13843876330611}
          cy={64}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
        <circle
          cx={39.42562584220406}
          cy={27.71281292110203}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
      </g>
      <g>
        <circle
          cx={55.42562584220407}
          cy={112}
          fill="white"
          r={9}
          stroke="#000"
          strokeWidth="2"
        />
        <text
          fill="black"
          fontSize="11"
          textAnchor="middle"
          x={55.42562584220407}
          y={116}
        >
          10
        </text>
      </g>
      <text
        fill="black"
        fontSize={8}
        textAnchor="end"
        x={109.85125168440814}
        y={94}
      >
        2.0
      </text>
    </svg>
    <svg
      height={128}
      viewBox="0 0 110.85125168440814 128"
      width={110.85125168440814}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="yellow"
        points="0,32 55.42562584220407,0 110.85125168440814,32 110.85125168440814,96 55.42562584220407,128 0,96"
      />
      <line
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
        x1={83.13843876330611}
        x2={27.71281292110201}
        y1={16.000000000000007}
        y2={111.99999999999999}
      />
      <path
        d="M-1.2983731993508297e-14 64 A 32 32 0 0,0 27.712812921102028 16"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <g>
        <circle
          cx={69.28203230275508}
          cy={40}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
        <circle
          cx={15.999999999999996}
          cy={59.712812921102035}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
      </g>
      <g>
        <circle
          cx={55.42562584220407}
          cy={112}
          fill="white"
          r={9}
          stroke="#000"
          strokeWidth="2"
        />
        <text
          fill="black"
          fontSize="11"
          textAnchor="middle"
          x={55.42562584220407}
          y={116}
        >
          10
        </text>
      </g>
      <text
        fill="black"
        fontSize={8}
        textAnchor="end"
        x={109.85125168440814}
        y={94}
      >
        2.1
      </text>
    </svg>
    <svg
      height={128}
      viewBox="0 0 110.85125168440814 128"
      width={110.85125168440814}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="yellow"
        points="0,32 55.42562584220407,0 110.85125168440814,32 110.85125168440814,96 55.42562584220407,128 0,96"
      />
      <line
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
        x1={27.712812921102046}
        x2={83.13843876330611}
        y1={16}
        y2={112}
      />
      <path
        d="M27.712812921102014 112 A 32 32 0 0,0 2.4008029108380828e-14 63.999999999999986"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <g>
        <circle
          cx={41.56921938165306}
          cy={40}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
        <circle
          cx={31.999999999999986}
          cy={96}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
      </g>
      <g>
        <circle
          cx={55.42562584220407}
          cy={112}
          fill="white"
          r={9}
          stroke="#000"
          strokeWidth="2"
        />
        <text
          fill="black"
          fontSize="11"
          textAnchor="middle"
          x={55.42562584220407}
          y={116}
        >
          10
        </text>
      </g>
      <text
        fill="black"
        fontSize={8}
        textAnchor="end"
        x={109.85125168440814}
        y={94}
      >
        2.2
      </text>
    </svg>
    <svg
      height={128}
      viewBox="0 0 110.85125168440814 128"
      width={110.85125168440814}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="yellow"
        points="0,32 55.42562584220407,0 110.85125168440814,32 110.85125168440814,96 55.42562584220407,128 0,96"
      />
      <line
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
        x1={110.85125168440814}
        x2={0}
        y1={64}
        y2={63.99999999999999}
      />
      <path
        d="M83.13843876330608 112 A 32 32 0 0,0 27.712812921102014 112"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <g>
        <circle
          cx={83.13843876330611}
          cy={64}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
        <circle
          cx={71.42562584220404}
          cy={100.28718707889796}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
      </g>
      <g>
        <circle
          cx={55.42562584220407}
          cy={112}
          fill="white"
          r={9}
          stroke="#000"
          strokeWidth="2"
        />
        <text
          fill="black"
          fontSize="11"
          textAnchor="middle"
          x={55.42562584220407}
          y={116}
        >
          10
        </text>
      </g>
      <text
        fill="black"
        fontSize={8}
        textAnchor="end"
        x={109.85125168440814}
        y={94}
      >
        2.3
      </text>
    </svg>
    <svg
      height={128}
      viewBox="0 0 110.85125168440814 128"
      width={110.85125168440814}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="yellow"
        points="0,32 55.42562584220407,0 110.85125168440814,32 110.85125168440814,96 55.42562584220407,128 0,96"
      />
      <line
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
        x1={83.13843876330611}
        x2={27.71281292110201}
        y1={16.000000000000007}
        y2={111.99999999999999}
      />
      <path
        d="M110.85125168440817 64.00000000000003 A 32 32 0 0,0 83.13843876330607 112"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <g>
        <circle
          cx={69.28203230275508}
          cy={40}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
        <circle
          cx={94.85125168440815}
          cy={68.28718707889797}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
      </g>
      <g>
        <circle
          cx={55.42562584220407}
          cy={112}
          fill="white"
          r={9}
          stroke="#000"
          strokeWidth="2"
        />
        <text
          fill="black"
          fontSize="11"
          textAnchor="middle"
          x={55.42562584220407}
          y={116}
        >
          10
        </text>
      </g>
      <text
        fill="black"
        fontSize={8}
        textAnchor="end"
        x={109.85125168440814}
        y={94}
      >
        2.4
      </text>
    </svg>
    <svg
      height={128}
      viewBox="0 0 110.85125168440814 128"
      width={110.85125168440814}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="yellow"
        points="0,32 55.42562584220407,0 110.85125168440814,32 110.85125168440814,96 55.42562584220407,128 0,96"
      />
      <line
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
        x1={27.712812921102046}
        x2={83.13843876330611}
        y1={16}
        y2={112}
      />
      <path
        d="M83.1384387633061 16 A 32 32 0 0,0 110.85125168440814 64"
        fill="none"
        stroke="black"
        strokeDasharray="1,0"
        strokeWidth={8}
      />
      <g>
        <circle
          cx={41.56921938165306}
          cy={40}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
        <circle
          cx={78.85125168440814}
          cy={31.999999999999996}
          fill="black"
          r={10}
          stroke="yellow"
          strokeWidth={2}
        />
      </g>
      <g>
        <circle
          cx={55.42562584220407}
          cy={112}
          fill="white"
          r={9}
          stroke="#000"
          strokeWidth="2"
        />
        <text
          fill="black"
          fontSize="11"
          textAnchor="middle"
          x={55.42562584220407}
          y={116}
        >
          10
        </text>
      </g>
      <text
        fill="black"
        fontSize={8}
        textAnchor="end"
        x={109.85125168440814}
        y={94}
      >
        2.5
      </text>
    </svg>
  </div>
</div>
`;
